docs/data/sql_functions.yml

```diff
@@ -634,6 +634,9 @@ collection:
   - sql: ARRAY_DISTINCT(haystack)
     table: haystack.arrayDistinct()
     description: Returns an array with unique elements. If the array itself is null, the function will return null. Keeps ordering of elements.
+  - sql: ARRAY_POSITION(haystack, needle)
+    table: haystack.arrayPosition(needle)
+    description: Returns the position of the first occurrence of element in the given array as long. Returns 0 if the given value could not be found in the array. Returns null if either of the arguments are null. And this is not zero based, but 1-based index. The first element in the array has index 1.
```

Comment:
Why not return `int` type? I see many vendors return `int` like PG/SnowFlake, etc. Also, it we can return `int` directly, then we won't need to cast in the function implementation.

Reply:
spark return bigint, just aligned with spark  https://github.com/apache/spark/blob/5a56c17283103821714ffaaf1c764e05d0ff6b58/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala#L2294

Reply:
and snowflake return is  0-based while spark is 1-based. 

https://docs.snowflake.com/en/sql-reference/functions/array_position

Reply:
and PG return int type and is also  1-based.

Reply:
TBH, either 0-based or 1-based is fine to me. But I still prefer return `int` type. The reason is as follows:
1: If we  return `int` type, we then won't need to extra cast in implementaion.
2: We don't need to always align with Spark. AFAICS, most of vendors return `int` type. So why not align most style of vendors.

Reply:
fixed

Reply:
fixed it